public/app/plugins/datasource/loki/configuration/QuerySettings.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import { config } from '@grafana/runtime';
+import { LegacyForms } from '@grafana/ui';
+
+const { FormField } = LegacyForms;
+
+type Props = {
+  maxLines: string;
+  onMaxLinedChange: (value: string) => void;
+  predefinedOperations: string;
+  onPredefinedOperationsChange: (value: string) => void;
+};
+
+export const QuerySettings = (props: Props) => {
+  const { maxLines, onMaxLinedChange, predefinedOperations, onPredefinedOperationsChange } = props;
+  return (
+    <>
+      <h3 className="page-heading">Queries</h3>
+      <div className="gf-form-group">
+        <div className="gf-form-inline">
+          <div className="gf-form">
+            <FormField
+              label="Maximum lines"
+              labelWidth={11}
+              inputWidth={20}
+              inputEl={
+                <input
+                  type="number"
+                  className="gf-form-input width-8 gf-form-input--has-help-icon"
+                  value={maxLines}
+                  onChange={(event) => onMaxLinedChange(event.currentTarget.value)}
+                  spellCheck={false}
+                  placeholder="1000"
+                />
+              }
+              tooltip={
+                <>
+                  Loki queries must contain a limit of the maximum number of lines returned (default: 1000). Increase
+                  this limit to have a bigger result set for ad-hoc analysis. Decrease this limit if your browser
+                  becomes sluggish when displaying the log results.
+                </>
+              }
+            />
+          </div>
+        </div>
+        {config.featureToggles.lokiPredefinedOperations && (
+          <div className="gf-form-inline">
```

Comment:
I guess we should somewhere add that this is experimental, WDYT?

Reply:
good idea!

Reply:
Updated in https://github.com/grafana/grafana/pull/69379/commits/875184d9bf20860b780261f42e97b4f11cf7da59

<img width="908" alt="image" src="https://github.com/grafana/grafana/assets/30407135/a09becfe-8b07-46cc-bd08-eb788afc95bc">